openshift/jenkins/jenkins-secondary.dc.yaml

```diff
@@ -0,0 +1,171 @@
+                  /var/run/secrets/jenkins-slave-user/password -description "$(cat
+                  /etc/hostname)" -executors ${SECONDARY_EXECUTORS} -labels '${SECONDARY_LABELS}'
+                  -mode 'normal' -retry 10 -tunnel $JENKINS_MASTER_SERVICE:50000
+                  -disableClientsUniqueId
+              env:
+                - name: JENKINS_MASTER_SERVICE
+                  value: ${NAME}
+                - name: JAVA_TOOL_OPTIONS
+                  value:
+                    -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
+                    -XX:MaxRAMFraction=5 -XX:MaxHeapFreeRatio=20 -XX:MinHeapFreeRatio=10
+                    -XX:+UseParallelGC -XX:ParallelGCThreads=2
+                - name: ENV_NAME
+                  value: "${ENV_NAME}"
+                - name: ENV_ID
+                  value: "${ENV_ID}"
+              imagePullPolicy: Always
+              name: jenkins
+              ports:
+                - containerPort: 50000
+                  protocol: TCP
+                - containerPort: 8080
+                  protocol: TCP
+              resources:
+                limits:
+                  cpu: "${CPU_LIMIT}"
+                  memory: "${MEMORY_LIMIT}"
+                requests:
+                  cpu: "${CPU_REQUEST}"
+                  memory: "${MEMORY_REQUEST}"
+              terminationMessagePath: /dev/termination-log
+              terminationMessagePolicy: File
+              volumeMounts:
+                - mountPath: /var/run/pod
+                  name: pod-metadata
+                - mountPath: /run/secrets/jenkins-slave-user
+                  name: jenkins-slave-user
+                  readOnly: true
+                - mountPath: /run/secrets/github
+                  name: github
+                  readOnly: true
+          dnsPolicy: ClusterFirst
+          restartPolicy: Always
+          schedulerName: default-scheduler
+          securityContext: {}
+          serviceAccount: ${NAME}
+          serviceAccountName: ${NAME}
+          terminationGracePeriodSeconds: 30
+          volumes:
+            # - name: jenkins-home
+            #   persistentVolumeClaim:
+            #     claimName: ${NAME}
+            - name: pod-metadata
+              downwardAPI:
+                items:
+                  - fieldRef:
+                      fieldPath: metadata.name
+                    name: name
+                    path: name
+                  - fieldRef:
+                      fieldPath: metadata.namespace
+                    name: namespace
+                    path: namespace
+                  - fieldRef:
+                      fieldPath: metadata.labels
+                    name: labels
+                    path: labels
+                  - fieldRef:
+                      fieldPath: metadata.annotations
+                    name: annotations
+                    path: annotations
+            - name: jenkins-slave-user
```

Comment:
are we back to using the term "slave"?

Reply:
It looks pretty tightly bound to what Jenkins needs to operate, so I left it alone.

Reply:
I'd love to get rid of Jenkins entirely, then the problem goes away...

Reply:
ah I see